### PR TITLE
Post tweet

### PR DESCRIPTION
**Pull request by JACornelius**

### CODE REVIEW

**Comment by kev-chan** [comment]
in main.js don't use global variables such as :
let homeTimelineResultClass;

File: src/js/components.js [hunk 1]
```diff
@@ -32,25 +38,35 @@ class User extends React.Component {
...
 class TimelineComponent extends React.Component {
 	render() {
 		if(this.props.timelineType == 'User') {
-			return e('div', {className: 'userTimelineContainer', key: 'userTimelineCont'},
-								 	[e('h2', {className: 'header', key: 'userTimelineHeader'}, 'User Timeline'),
-									 e(TimelineButton, {className: 'userTimelineButton', key: 'userTimelineButt',onClickFunc: this.props.buttonFunc, buttonText: 'Get User Timeline'}, null),
-									 e(TimelineResultComp, {timelinePlaceholder: 'userTimelinePlaceholder', className: this.props.resultClass, key: 'userTimelineResComponent', timelineResult: this.props.resultOutput}, null)]);
+			return e('div', {id: 'userTimelineContainer', key: 'userTimelineCont'},
+					 	[e('h2', {className: 'header', key: 'userTimelineHeader'}, 'User Timeline'),
+						 e(ButtonComponent, {className: 'userTimelineButton', key: 'userTimelineButt',onClickFunc: this.props.buttonFunc, buttonText: 'Get User Timeline'}, null),
```
Comment (kev-chan) [inline src/js/components.js]: you don't need the trailing null (at least i'm pretty sure you don't)

File: src/js/components.js [hunk 2]
```diff
@@ -32,25 +38,35 @@ class User extends React.Component {
...
+						 e(TimelineResultComponent, {timelinePlaceholder: 'userTimelinePlaceholder', className: this.props.resultClass, key: 'userTimelineResComponent', timelineResult: this.props.resultOutput}, null)]);
```
Comment (kev-chan) [inline src/js/components.js]: same as above, javascript will default it to 'undefined'

File: src/js/components.js [hunk 3]
```diff
@@ -32,25 +38,35 @@ class User extends React.Component {
...
 		}
 		else {
-			return e('div', {className: 'homeTimelineContainer', key: 'homeTimelineCont'}, 
+			return e('div', {id: 'homeTimelineContainer', key: 'homeTimelineCont'}, 
 						[e('h2', {className: 'header', key: 'homeTimelineHeader'}, 'Home Timeline'),
-						 e(TimelineButton, {className: 'filterButton', disabledFilter: !this.props.filter, id: 'filterButton', onClickFunc: this.props.filterButtonFunc, buttonText: 'Filter'}, null),
-						 e(FilterInputBox, {onEnter: this.props.onKeyPressButton, onChangeFilter: this.props.onChangeButton, filterState: this.props.filter}, null),
-						 e(TimelineButton, {className: 'homeTimelineButton', key: 'homeTimeLineButt', onClickFunc: this.props.buttonFunc, buttonText: 'Get Home Timeline'}, null),
-						 e(TimelineResultComp, {timelinePlaceholder: 'homeTimelinePlaceholder', className: this.props.resultClass, key: 'homeTimelineResComponent', timelineResult: this.props.resultOutput}, null)]);
+						 e(ButtonComponent, {className: 'filterButton', disabledButton: !this.props.filter, id: 'filterButton', onClickFunc: this.props.filterButtonFunc, buttonText: 'Filter'}, null),
```
Comment (kev-chan) [inline src/js/components.js]: same as above regarding nulls for the following lines

File: src/js/services.js [hunk 4]
```diff
@@ -45,4 +45,15 @@ const getFilterTimeline = (filter) => {
 	return fetch('http://localhost:8080/api/1.0/twitter/tweet/filter?filter=' + filter)
 }
 
-export{getHomeTimeline, getUserTimeline, getFilterTimeline, renderedTimeline};
\ No newline at end of file
+const postTweet = (tweet) => {
+	let data = {"name": "mojo", "message": tweet};
```
Comment (kev-chan) [inline src/js/services.js]: why is name 'mojo'?

File: src/sass/style.scss [hunk 5]
```diff
@@ -55,81 +56,136 @@
...
 			@include alternatingColors(#f5f8fa, #1da1f2)
 		}
 	}
 }
 
-.Timelines{
-	width: 1500px;
-	margin: auto;
-	.homeTimelineContainer, .userTimelineContainer {
+textarea {
```
Comment (kev-chan) [inline src/sass/style.scss]: don't use element names

File: src/sass/style.scss [hunk 6]
```diff
@@ -55,81 +56,136 @@
...
-					float: right;
-					margin: auto;
-					text-align: left;
-
-					.dateBlock {
-						font-size: 12px;
-						padding-bottom: 5px;
-						font-style: italic;
-					}	
+	&__tab-panel {
```
Comment (kev-chan) [inline src/sass/style.scss:127]: looks like your } may be misplaced

File: src/sass/style.scss [hunk 7]
```diff
@@ -55,81 +56,136 @@
...
+				margin: auto;
+				text-align: left;
+
+				.dateBlock {
+					font-size: 12px;
+					padding-bottom: 5px;
+					font-style: italic;
+				}	
+			}
+			.User {
```
Comment (kev-chan) [inline src/sass/style.scss]: stick with camel case for all class names

File: src/sass/style.scss [hunk 8]
```diff
@@ -55,81 +56,136 @@
...
+				padding: 10px 0px 0px 10px;
```
Comment (kev-chan) [inline src/sass/style.scss]: simplify css rule

File: src/js/main.js [hunk 9]
```diff
@@ -120,35 +220,37 @@ class Timelines extends React.Component {
...
+}
+
+class TabsAndTabPages extends React.Component {
+	render() {
+		return e(Tabs, {}, [
+					e(TabList, {},[
+						e(Tab, {label: "homeTimelineTab"}, "Home Timeline"),
+						e(Tab, {label: "userTimelineTab"}, "User Timeline"),
+						e(Tab, {label: "postTweetTab"}, "Post Tweet")]),
+					e(TabPanel, {}, e(HomeTimeline, {}, null)),
```
Comment (kev-chan) [inline src/js/main.js]: simplify, you aren't passing props or element so just create the element

File: src/js/main.js [hunk 10]
```diff
@@ -120,35 +220,37 @@ class Timelines extends React.Component {
...
+					e(TabPanel, {}, e(UserTimeline, {}, null)),
```
Comment (kev-chan) [inline src/js/main.js]: same here and below

File: src/js/main.js [hunk 11]
```diff
@@ -120,35 +220,37 @@ class Timelines extends React.Component {
...
 								  buttonFunc: () => {this.handleGetUserTimeline()}, 
 							 	  resultClass: userTimelineResultClass, 
-							 	  resultOutput:userTimelineResultOutput}, null)]);			
+							 	  resultOutput: userTimelineResultOutput}, null);	
+	}
+}
+
+class TabsAndTabPages extends React.Component {
+	render() {
+		return e(Tabs, {}, [
```
Comment (kev-chan) [inline src/js/main.js:266]: cool that you're using a component vs building your own

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: src/js/main.js (hunk 9 above)
Comment (JACornelius) [reply src/js/main.js]: i need the label for selenium, for getting the tab

File: src/js/services.js (hunk 4 above)
Comment (JACornelius) [reply src/js/services.js]: it was just defualt name, ill change it to "default name"

File: src/sass/style.scss (hunk 8 above)
Comment (JACornelius) [reply src/sass/style.scss]: its only top and bottom

File: src/js/services.js (hunk 4 above)
Comment (JACornelius) [reply src/js/services.js]: nvm I just didnt pass anything for the name attribute

File: __tests__/PostTweetComponentTest.js [hunk 12]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import {shallow, mount, render} from 'enzyme';
+import {shallowToJson} from 'enzyme-to-json';
+import {PostTweetComponent, TweetInput, ButtonComponent} from '../src/js/components';
+
+let count = 0;
```
Comment (kev-chan) [inline __tests__/PostTweetComponentTest.js]: why is there a testCounter variable?

File: src/js/main.js [hunk 13]
```diff
@@ -88,67 +69,217 @@ class Timelines extends React.Component {
...
+			value: event.target.value
+		})
+	}
+	
+	handlePostTweet(message) {
+		postTweet(message).then(response => response.json())
+						  .then(data => {
+						  	this.setState ({
+						  		postTweetMessage: data.message,
+						  		postTweetError: false
```
Comment (kev-chan) [inline src/js/main.js]: for boolean types, isPostTweetError would be better name
postTweetMessage being a message and postTweetError being a boolean is a bit confusion

Comment (kev-chan) [reply src/js/main.js]: in general booleans using 'is<Whatever>' works well

File: src/js/main.js [hunk 14]
```diff
@@ -88,67 +69,217 @@ class Timelines extends React.Component {
...
+}
+
+class TabsAndTabPages extends React.Component {
+	render() {
+		return e(Tabs, {}, [
+					e(TabList, {},[
+						e(Tab, {label: "homeTimelineTab"}, "Home Timeline"),
+						e(Tab, {label: "userTimelineTab"}, "User Timeline"),
+						e(Tab, {label: "postTweetTab"}, "Post Tweet")]),
+					e(TabPanel, {}, e(HomeTimeline, {})),
```
Comment (kev-chan) [inline src/js/main.js]: do you need the {} for the components?  (HomeTimeline/UserTimeline/PostTweet)

File: src/sass/style.scss [hunk 15]
```diff
@@ -25,28 +31,34 @@
 	font-weight: bold;
 }
 
+.successPostTweet {
```
Comment (kev-chan) [inline src/sass/style.scss:34]: any reason why the success/error have different right positions?  If they are the same element, use can use a shared class for shared css and then use the additional class to change the color

File: src/sass/style.scss [hunk 16]
```diff
@@ -55,81 +67,139 @@
...
-		#homeTimelinePlaceholder, #userTimelinePlaceholder {
-			margin-top: 10px;
-			width: 95%;
-			border: {
-				style: solid;
-				width: 1px;
-			}
+		list-style: none;
+		padding: 6px 12px;
+		cursor: pointer;
```
Comment (kev-chan) [inline src/sass/style.scss:117]: nice having it be pointer

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: src/js/main.js (hunk 14 above)
Comment (JACornelius) [reply src/js/main.js]: only if there are components inside the created react element but ill remove instances taht are not that case

File: src/sass/style.scss [hunk 17]
```diff
@@ -55,81 +69,139 @@
...
+		&--selected {
+			background: maroon;
+			color: black;
+		}
+	}
+
+	&__tab-panel {
+		display: none;
+
+	&--selected {
```
Comment (kev-chan) [inline src/sass/style.scss]: fix whitespace

**Review by kev-chan — CHANGES_REQUESTED** [review]
please fix whitespacing in the scss file.  Otherwise things look good and you can merge after the whitespace i fixed;

**Pull request merged**